sdk/lib/cmlib/cmcheck.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * PROJECT:         ReactOS Registry Manipulation Library
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Registry Validation
+ * COPYRIGHT:       Copyright 2022 George Bișoc <george.bisoc@reactos.org>
+ */
+
+/* INCLUDES ******************************************************************/
+
+#include "cmlib.h"
+#define NDEBUG
+#include <debug.h>
+
+/* DEFINES  ******************************************************************/
+
+#define GET_HHIVE(CmHive)   (&((CmHive)->Hive))
+
+#if !defined(CMLIB_HOST) && !defined(_BLDR_)
+extern PCMHIVE CmiVolatileHive;
+#endif
+
+/* PUBLIC FUNCTIONS ***********************************************************/
+
+ULONG
+NTAPI
+CmCheckRegistry(
+    _In_ PCMHIVE RegistryHive,
+    _In_ ULONG Flags)
+{
+    ULONG CmStatusCode;
+    PHHIVE Hive;
+
+    /* Make sure the submitted flag is valid */
+    ASSERT(Flags & CM_CHECK_REGISTRY_VALID_FLAGS);
+
+#if !defined(CMLIB_HOST) && !defined(_BLDR_)
+    /*
+     * If this CM hive that we got is a volatile
+     * hive (aka master hive which is what it's called
+     * on Windows) then registry validation checks is
+     * not necessary.
+     */
+    if (RegistryHive == CmiVolatileHive)
+    {
+        DPRINT("CmCheckRegistry(): This is master registry hive, don't do anything!\n");
+        return CM_CHECK_REGISTRY_GOOD;
+    }
+#endif
+
+    /*
+     * Obtain the hive and check if the caller wants
+     * that the hive has to be validated.
+     */
+    Hive = GET_HHIVE(RegistryHive);
+
+    UNREFERENCED_PARAMETER(CmStatusCode);
+    UNREFERENCED_PARAMETER(Hive);
```

Comment:
These two lines have been added to fix the build for now. When you implement the "TODO" and therefore, use those variables, these `UNREFERENCED_PARAMETER` should go away.